networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/core/encodings/ssz/BeaconBlocksByRootRequestMessageEncoder.java

```diff
@@ -44,6 +46,10 @@ public BeaconBlocksByRootRequestMessage decode(final Bytes message) throws RpcEx
     for (int i = 0; i < message.size(); i += Bytes32.SIZE) {
       blockRoots.add(Bytes32.wrap(message.slice(i, Bytes32.SIZE)));
     }
-    return new BeaconBlocksByRootRequestMessage(blockRoots);
+    try {
+      return new BeaconBlocksByRootRequestMessage(blockRoots);
+    } catch (IllegalArgumentException ex) {
+      throw new RpcException(INVALID_REQUEST_CODE, ex.getMessage());
+    }
```

Comment:
We wouldn't need this try/catch if the constructor just threw RpcException.  Ultimately we should catch this once the length prefix is read because the message length can't be more than Bytes32.SIZE * MAX_REQUEST_BLOCKS - there's a separate ticket for that though.